components/proxy/src/main/java/com/hotels/styx/admin/AdminServerBuilder.java

```diff
@@ -82,68 +86,104 @@
+                .add("/admin/threads", new ThreadsHandler())
+                .add("/admin/metrics", new MetricsHandler(environment.metricRegistry(), metricsCacheExpiration))
+                .add("/admin/healthcheck", healthCheckHandler)
+                .add("/admin/configuration", new StyxConfigurationHandler(staticConfiguration()))
+                .add("/admin/configuration/origins", new OriginsHandler(backendServicesRegistry))
+                .add("/admin/jvm", new JVMMetricsHandler(environment.metricRegistry(), metricsCacheExpiration))
+                .add("/admin/origins/status", new OriginsInventoryHandler(environment.eventBus()))
+                .add("/admin/configuration/logging", new LoggingConfigurationHandler(styxConfig.startupConfig().logConfigLocation()))
+                .add("/admin/configuration/startup", new StartupConfigHandler(styxConfig.startupConfig()))
+                .add("/admin/styx/proxy/status", new ProxyStatusHandler(environment.configStore()))
+
+                // Dashboard
+                .add("/admin/dashboard/data.json", dashboardDataHandler(styxConfig))
+                .add("/admin/dashboard/", new ClassPathResourceHandler("/admin/dashboard/"))
+
+                // Tasks
+                .add("/admin/tasks/origins/reload", new HttpMethodFilteringHandler(POST, new OriginsReloadCommandHandler(backendServicesRegistry)))
+                .add("/admin/tasks/origins", new HttpMethodFilteringHandler(POST, new OriginsCommandHandler(environment.eventBus())))
+                ;
+
+        environment.configStore().watch("plugins", List.class)
+                .map(list -> (List<NamedPlugin>) list)
+                .subscribe(plugins -> {
+                    routesForPlugins(plugins).forEach(route -> httpRouter.add(route.path(), route.handler()));
+                    httpRouter.add("/admin/tasks/plugin/", new PluginToggleHandler(plugins));
+                    httpRouter.add("/admin/plugins", new PluginListHandler(plugins));
+                });
+
+        verifyThatLinksInIndexMapToRealEndpoints(links, httpRouter.paths);
+
+        return new NettyServerBuilderSpec("admin", environment.serverEnvironment(), new WebServerConnectorFactory())
                 .toNettyServerBuilder(adminServerConfig)
-                .httpHandler(httpRouter)
+                .httpHandler(() -> httpRouter.router)
+                .configStore(environment.configStore())
                 .build();
     }
 
+    private static final class RouterBuilder {
+        private final StandardHttpRouter router = new StandardHttpRouter();
+        private final Set<String> paths = new HashSet<>();
+
+        public RouterBuilder add(String path, HttpHandler2 handler) {
+            this.router.add(path, handler);
+            this.paths.add(path);
+            return this;
+        }
+    }
+
+    private static void verifyThatLinksInIndexMapToRealEndpoints(Iterable<IndexHandler.Link> links, Set<String> paths) {
```

Comment:
Isn't this something we should have in the unit/e2e tests, rather than in the running code?

What would be the suggested remedial action for the end user if this exception ever fires?